### PR TITLE
refactor: Minimize `Vector2` creation in `IsometricTileMapComponent`

### PR DESCRIPTION
**Pull request by spydon**
# Description

The `IsometricTileMapComponent` was creating a lot of new `Vector2` objects, this PR minimizes the creation of them.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: packages/flame/lib/src/extensions/vector2.dart [hunk 1]
```diff
@@ -175,6 +175,12 @@ extension Vector2Extension on Vector2 {
     }
   }
 
+  /// Adds [x] and [y] to the current [Vector2].
+  void addValues(double x, double y) {
```
Comment (ufrshubham) [inline packages/flame/lib/src/extensions/vector2.dart:179]: Isn't that same as `translate`?

Comment (spydon) [reply packages/flame/lib/src/extensions/vector2.dart:179]: Aaah, of course, good catch :)

**Review by ufrshubham — APPROVED** [review]

File: packages/flame/lib/src/extensions/vector2.dart [hunk 2]
```diff
@@ -175,6 +175,12 @@ extension Vector2Extension on Vector2 {
...
+    this.x += x;
+    this.y += y;
+  }
+
```
Comment (spydon) [inline packages/flame/lib/src/extensions/vector2.dart:183]:
```suggestion
```

File: packages/flame/test/extensions/vector2_test.dart [hunk 3]
```diff
@@ -472,6 +472,12 @@ void main() {
       expect(p1.hashCode == p2.hashCode, false);
     });
 
+    test('addValues', () {
+      final p = Vector2(1.0, 0.0);
+      p.addValues(2.0, 3.0);
+      expect(p, Vector2(3.0, 3.0));
+    });
+
```
Comment (spydon) [inline packages/flame/test/extensions/vector2_test.dart:480]:
```suggestion
```

File: packages/flame/lib/src/components/isometric_tile_map_component.dart [hunk 4]
```diff
@@ -147,29 +154,42 @@ class IsometricTileMapComponent extends PositionComponent {
...
   /// This is the opposite of [getBlockRenderPosition].
   Block getBlockRenderedAt(Vector2 p) {
     final tile = effectiveTileSize;
     return getBlock(
-      p +
-          (Vector2(tile.x / 2, tile.y - effectiveTileHeight - tile.y / 4)
-            ..multiply(scale)),
+      _blockPositionCache
+        ..setFrom(p)
+        ..addValues(
```
Comment (spydon) [inline packages/flame/lib/src/components/isometric_tile_map_component.dart:189]:
```suggestion
        ..translate(
```

File: packages/flame/lib/src/components/isometric_tile_map_component.dart [hunk 5]
```diff
@@ -147,29 +154,42 @@ class IsometricTileMapComponent extends PositionComponent {
...
+    final halfTile = _getBlockCache
+      ..setFrom(effectiveTileSize)
+      ..multiply(scale / 2);
     final multiplier = 1 - halfTile.y / (2 * effectiveTileHeight * scale.x);
-    final delta = halfTile.clone()..multiply(Vector2(1, multiplier));
-    final cart = isoToCart(p - position + delta);
+    final iso = _getBlockIsoCache
+      ..setFrom(p)
+      ..sub(position)
+      ..addValues(halfTile.x, halfTile.y * multiplier);
```
Comment (spydon) [inline packages/flame/lib/src/components/isometric_tile_map_component.dart:172]:
```suggestion
      ..translate(halfTile.x, halfTile.y * multiplier);
```

File: packages/flame/lib/src/components/isometric_tile_map_component.dart [hunk 6]
```diff
@@ -127,10 +133,11 @@ class IsometricTileMapComponent extends PositionComponent {
   /// This is the opposite of [getBlock].
   Vector2 getBlockCenterPosition(Block block) {
     final tile = effectiveTileSize;
-    final result = getBlockRenderPosition(block) +
-        (Vector2(tile.x / 2, tile.y - effectiveTileHeight - tile.y / 4)
-          ..multiply(scale));
-    return result;
+    return getBlockRenderPosition(block)
+      ..addValues(
```
Comment (spydon) [inline packages/flame/lib/src/components/isometric_tile_map_component.dart:137]:
```suggestion
      ..translate(
```

**Pull request merged**